### DISCUSSION TITLE
Social logins etc

### DISCUSSION THREAD

**Discussion by igmltd** [rapidez/rapidez#92] [General]
Just a general thought

If an integration required social logins, how would be possible to login the user in magento - the below requires the password?
  
```
export const login = async function (email, password) {
    return magentoGraphQL(
        'mutation generateCustomerToken ($email: String!, $password: String!) { generateCustomerToken (email: $email, password: $password) { token } }',
        {
            email: email,
            password: password,
        },
    ).then(async (response) => {
        await loginByToken(response.data.generateCustomerToken.token)
        return response
    })
}
```
How would this be approached?, is there a way to generate a customer token based on just email or id? 

Thanks

**Comment by royduin** [comment]
@justbetter did a Rapidez social login integration with Laravel Socialite. The approach there was to use the `generateCustomerTokenAsAdmin` GraphQL mutation from the PHP side as it requires a integration token: https://developer.adobe.com/commerce/webapi/get-started/authentication/gs-authentication-token/#integration-tokens

**Reply by igmltd** [reply]
Thanks for pointing me in the right direction, thats great 👍
